src/_path.h

```diff
@@ -872,6 +872,64 @@ bool path_intersects_path(PathIterator1 &p1, PathIterator2 &p2)
     return false;
 }
 
+// returns whether the segment from (x1,y1) to (x2,y2)
+// intersects the rectangle centered at (cx,cy) with size (w,h)
+inline bool segment_intersects_rectangle(double x1, double y1,
```

Comment:
(Absent further opinions, I think a reasonable option is to create a `src/doc` folder and put it there.  It only needs to be accessible from a source checkout anyways.)